### PR TITLE
work for PHP 8

### PR DESCRIPTION
**Pull request by remicollet**
@m6w6 please review

### CODE REVIEW

**Comment by remicollet** [comment]
PHP_VERSION : 8.0.0beta3

```
TEST RESULT SUMMARY
---------------------------------------------------------------------
Exts skipped    :    0
Exts tested     :   16
---------------------------------------------------------------------

Number of tests :   24                15
Tests skipped   :    9 ( 37.5%) --------
Tests warned    :    0 (  0.0%) (  0.0%)
Tests failed    :    0 (  0.0%) (  0.0%)
Tests passed    :   15 ( 62.5%) (100.0%)
---------------------------------------------------------------------
Time taken      :    0 seconds
=====================================================================

```

File: oauth.c [hunk 1]
```diff
@@ -249,7 +255,7 @@ zend_string *soo_sign_rsa(php_so_object *soo, char *message, const oauth_sig_con
 	ZVAL_NULL(&args[1]);
 	ZVAL_DUP(&args[2], &ctx->privatekey);
 
-	call_user_function_ex(EG(function_table), NULL, &func, &retval, 3, args, 0, NULL);
+	call_user_function(EG(function_table), NULL, &func, &retval, 3, args);
```
Comment (remicollet) [inline oauth.c:258]: Hope this one will not create issue (as no_separation=1 is now the default)

File: oauth.c [hunk 2]
```diff
@@ -1797,11 +1811,13 @@ SO_METHOD(__construct)
...
-	zend_update_property_null(soo_class_entry, obj, "debugInfo", sizeof("debugInfo") - 1);
-	zend_update_property_bool(soo_class_entry, obj, "debug", sizeof("debug") - 1, soo->debug);
-	zend_update_property_long(soo_class_entry, obj, "sslChecks", sizeof("sslChecks") - 1, soo->sslcheck);
+	zend_update_property_null(soo_class_entry, OBJ_FOR_PROP(obj), "debugInfo", sizeof("debugInfo") - 1);
+	zend_update_property_bool(soo_class_entry, OBJ_FOR_PROP(obj), "debug", sizeof("debug") - 1, soo->debug);
+	zend_update_property_long(soo_class_entry, OBJ_FOR_PROP(obj), "sslChecks", sizeof("sslChecks") - 1, soo->sslcheck);
 
+#if PHP_VERSION_ID < 80000
 	TSRMLS_SET_CTX(soo->thread_ctx);
+#endif
```
Comment (remicollet) [inline oauth.c:1820]: perhaps this can be safely removed, even with 7.x

**Comment by m6w6** [comment]
Your test run skips too many essential tests :wink: ; exts needed: openssl, posix, pcntl

**Comment by m6w6** [comment]
LGTM, now. 100% pass, 0% skip on 8 (master) and 7.1

**Review by m6w6 — APPROVED** [review]

**Comment by remicollet** [comment]
Thanks a lot, tested with 7.0, 7.1, 7.2, 7.3, 7.4 and 8.0.0beta4, build and test suite OK everywhere (without any skipped tests)

New version welcome ;)

**Pull request merged**